serial.c

```diff
@@ -182,12 +182,34 @@ struct ios_ops * serial_init(char *device)
 	fd = open(lockfile, O_RDWR | O_CREAT | O_EXCL, 0444);
 	if (fd < 0) {
 		if (errno == EEXIST) {
+			char pidbuf[12];
+			ssize_t nbytes = 0;
 			if (opt_force) {
 				printf("lockfile for port exists, ignoring\n");
 				serial_unlock();
 				goto relock;
 			}
 
+			fd = open(lockfile, O_RDONLY);
+			if (fd < 0)
+				main_usage(3, "lockfile for port can't be opened", device);
+
+			do {
+				ret = read(fd, &pidbuf[nbytes], sizeof(pidbuf) - nbytes - 1);
+				nbytes += ret;
+			} while (ret > 0 && nbytes < sizeof (pidbuf) - 1);
```

Comment:
Would it make sense here to do some additional checks on the format? E.g. check you read the file completely? If the lockfile happens to be created by an older microcom with pid 33333333 you check for pid 3333 instead. (I think, didn't test.)